cmd/autoscaler/app/autoscaler.go

```diff
@@ -75,15 +83,18 @@ func createAutoScaler(restConfig *rest.Config,
 		return nil, errors.Wrap(err, "Failed to initialize root logger")
 	}
 
-	customMetricsClient, err := custommetricsv1.NewForConfig(restConfig)
+	dc, err := discovery.NewDiscoveryClientForConfig(restConfig)
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed create custom metrics client set")
+		return nil, err
 	}
+	availableAPIsGetter := custom_metrics.NewAvailableAPIsGetter(dc)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(dc))
+	customMetricsClient := custom_metrics.NewForConfig(restConfig, mapper, availableAPIsGetter)
```

Comment:
All this to get `customerMetricsClient`?
I have no idea if there's an easier shorter way, so I'm counting on you there isn't

Reply:
Indeed. After investigation of mine and @Hedingber - this is the way we've found to create the `customMetricsClient` , given the fact we have `restConfig`